### ISSUE TITLE
Regarding `stripe_id` in `users` table

### ISSUE THREAD

**Issue by jorioux** [l4nos/laravel-cashier-stripe-connect#41]
Hi @l4nos , I'm planning on using your package extensively for my SaaS platform:

https://pickletime.app

I'm already using tenancyforlaravel v3, fully working. I want to use `laravel/cashier` and your package to allow tenants (pickleball clubs), to bill their members.

What's working so far:

* The tenants can create their account (connected account), and do the onboarding:

  ```php
  tenant()->createOrGetStripeAccount();
  $this->redirect(tenant()->accountDashboardUrl());
  ```
  
* Once the tenant has onboarded with Stripe, the members of the tenant can create their Stripe account:

  ```php
  $user->createStripeCustomer(tenant());
  ```

Now I want to allow the member to add payment methods by using:

```php
$this->redirect($user->billingPortalUrl($returnUrl));
```

But it looks like the `stripe_id` in the `users` table is not set after, and `billingPortalUrl()` requires it.

I'm I missing something? Thanks!

**Comment by l4nos** [comment]
Hi Jonathan,

Thanks for sharing this. I will look into this now and run a test in my boilerplate environment and come back to you.

Nice to see another fan of Tenancy for Laravel! (I'm a huge advocate of Archtec/Stancl's work).

Will come back with a response shortly.

**Comment by jorioux** [comment]
I looked at the source code for a while, and it seems I would have to implement the billingPortalUrl() function myself in the ManageCustomer file?

**Comment by l4nos** [comment]
Yes, the customer portal stuff is quite new in stripe, and tis library hasn't quite caught up with it yet.

Feel frre to fork and PR if you want.

Alternatively, I plan to do some updates over christmas. I've not had an opportunity thus far due to paid work coommitments.

**Comment by thumbtech** [comment]
Did you get anywhere? Have you tried working "directly" through the API? Something like:

```
$portal = Cashier::stripe()->billingPortal->sessions->create([
  'customer' => $user->stripeCustomerId(),
  'return_url' => 'https://example.com/account',
],
tenant()->stripeAccountOptions([], true)
);
$this->redirect($portal->url);
```

I'm going to be something similar so curious whether it works!

https://docs.stripe.com/api/customer_portal/sessions/create

**Comment by jorioux** [comment]
Yes that exactly what I did. 

I overridden some laravel/cashier files in my project to add the connect account to the request. 

It basically renders this package useless. I'm even using laravel/cashier to receive web hooks from stripe connect events. It would be so easy to add the stripe connect support in cashier, I don't know why they don't just add it.

**Comment by thumbtech** [comment]
Yes it is a matter of understanding the division of labor between this package, Cashier, Stripe API, and Stripe itself (checkout, portal, etc.)

I agree Cashier could add connect pretty easily, but that would be a distraction/brain-fart for many folks. Meanwhile I suppose we could assist with some work via PR's here?

I myself had to build custom OAuth connect onboarding, as I find that much better workflow is for customers to create their Stripe accounts first then connect to that account rather than this onboarding approach. That would be a good PR I think.

Hey how does your system compare to clublocker, which I'm pretty familiar with?

**Comment by l4nos** [comment]
Hi Gents, just interjecting here.

I agree with your points around Connect becoming part of Cashier, that would have been the ideal solution, really speaking this package should never had to have existed, but alas!

I haven't had time to work on this package recently due to commercial work taking up my time.

I'm happy to review and accept any valid PR requests. Also happy to update the documentation to reflect any changes on GItBook.

Appreciaite any support or contributions!

**Comment by thumbtech** [comment]
@l4nos something that concerns me here is the direct use of the stripe-php library throughout the package instead of leveraging Cashier, which otherwise provides complete access to the stripe client with appropriate configuration. For example, I noticed that stripe calls from laravel-cashier-stripe-connect were using a different API version than Cashier because stripeAccountOptions() method drops stripe_version (and also api_base) from the options stack. (At least these should be added there for consistency, I believe ... ?)

Perhaps most elegant would be to just route all calls through Cashier?

So something like following from CanCharge trait

```
return PaymentIntent::create($options, $this->stripeAccountOptions([],true));
```

may be more robust if like

```
return Cashier::stripe(['stripe_account' => $this->stripeAccountId()])->paymentIntents->create($options);
```

Anyway, just sort of gauging your appetite for this level of update presented in a PR, or if you are philosophically opposed somehow.

**Comment by l4nos** [comment]
Definitely not philosophically opposed, I built this library from another one that just handled basic onboarding links, and it used the same direct SDK access, but you're right, if developers want consistency and centralised control of API versioning, the plugin should leverage Cashiers implementation of the stripe interface.

Happy to review a PR.
